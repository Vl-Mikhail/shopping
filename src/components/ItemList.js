import React from "react";
import { Table } from "reactstrap";

export default ({ items }) => (
  <Table>
    <thead>
      <tr>
        <th>Название</th>
        <th>Цвет</th>
        <th>Цена</th>
      </tr>
    </thead>
    <tbody>
      {items.map((item, index) => {
        return (
          <tr key={index}>
            <td>{item.name}</td>
            <td>{item.color}</td>
            <td>{item.price}$</td>
          </tr>
        );
      })}
    </tbody>
  </Table>
);
