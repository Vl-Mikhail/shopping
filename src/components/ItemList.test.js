import React from "react";
import { shallow } from "enzyme";
import ItemList from "./ItemList";

const items = [
  {
    name: "Сандали",
    color: "зеленый",
    price: "23"
  }
];

it("renders without crashing", () => {
  expect(shallow(<ItemList items={items}/>)).toMatchSnapshot();
});
