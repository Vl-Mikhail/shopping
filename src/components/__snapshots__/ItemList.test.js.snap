// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    items={
        Array [
            Object {
              "color": "зеленый",
              "name": "Сандали",
              "price": "23",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <thead>
          <tr>
                    <th>
                              Название
                    </th>
                    <th>
                              Цвет
                    </th>
                    <th>
                              Цена
                    </th>
          </tr>
</thead>,
        <tbody>
          <tr>
                    <td>
                              Сандали
                    </td>
                    <td>
                              зеленый
                    </td>
                    <td>
                              23
                              $
                    </td>
          </tr>
</tbody>,
      ],
      "responsiveTag": "div",
      "tag": "table",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <tr>
            <th>
                        Название
            </th>
            <th>
                        Цвет
            </th>
            <th>
                        Цена
            </th>
</tr>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <th>
                Название
</th>,
              <th>
                Цвет
</th>,
              <th>
                Цена
</th>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Название",
              },
              "ref": null,
              "rendered": "Название",
              "type": "th",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Цвет",
              },
              "ref": null,
              "rendered": "Цвет",
              "type": "th",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Цена",
              },
              "ref": null,
              "rendered": "Цена",
              "type": "th",
            },
          ],
          "type": "tr",
        },
        "type": "thead",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <tr>
              <td>
                            Сандали
              </td>
              <td>
                            зеленый
              </td>
              <td>
                            23
                            $
              </td>
</tr>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <td>
                  Сандали
</td>,
                <td>
                  зеленый
</td>,
                <td>
                  23
                  $
</td>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Сандали",
                },
                "ref": null,
                "rendered": "Сандали",
                "type": "td",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "зеленый",
                },
                "ref": null,
                "rendered": "зеленый",
                "type": "td",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "23",
                    "$",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "23",
                  "$",
                ],
                "type": "td",
              },
            ],
            "type": "tr",
          },
        ],
        "type": "tbody",
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <thead>
            <tr>
                        <th>
                                    Название
                        </th>
                        <th>
                                    Цвет
                        </th>
                        <th>
                                    Цена
                        </th>
            </tr>
</thead>,
          <tbody>
            <tr>
                        <td>
                                    Сандали
                        </td>
                        <td>
                                    зеленый
                        </td>
                        <td>
                                    23
                                    $
                        </td>
            </tr>
</tbody>,
        ],
        "responsiveTag": "div",
        "tag": "table",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <tr>
              <th>
                            Название
              </th>
              <th>
                            Цвет
              </th>
              <th>
                            Цена
              </th>
</tr>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <th>
                  Название
</th>,
                <th>
                  Цвет
</th>,
                <th>
                  Цена
</th>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Название",
                },
                "ref": null,
                "rendered": "Название",
                "type": "th",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Цвет",
                },
                "ref": null,
                "rendered": "Цвет",
                "type": "th",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Цена",
                },
                "ref": null,
                "rendered": "Цена",
                "type": "th",
              },
            ],
            "type": "tr",
          },
          "type": "thead",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <tr>
                <td>
                                Сандали
                </td>
                <td>
                                зеленый
                </td>
                <td>
                                23
                                $
                </td>
</tr>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <td>
                    Сандали
</td>,
                  <td>
                    зеленый
</td>,
                  <td>
                    23
                    $
</td>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Сандали",
                  },
                  "ref": null,
                  "rendered": "Сандали",
                  "type": "td",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "зеленый",
                  },
                  "ref": null,
                  "rendered": "зеленый",
                  "type": "td",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "23",
                      "$",
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    "23",
                    "$",
                  ],
                  "type": "td",
                },
              ],
              "type": "tr",
            },
          ],
          "type": "tbody",
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
